DataService.js

```diff
@@ -85,6 +87,28 @@ angular.module('PathOfDamage')
         });
       }
       return table;
+    },
+    generateHitList: function (hits) {
+      if (!hits) {
+        return null;
+      }
+      var list = [];
+      for (var i = 0; i < hits.length; i++) {
+        list.push(hits[i].hit);
+      }
+      return list;
+    },
+    generateHitObject: function (hitList) {
+      if (!hitList) {
```

Comment:
what if we restructured to if (hitList) { ...method }. Does not returning anything break anything? That makes the return undefined instead of null correct?